application/shared-kernel/ApiCore/Middleware/WebAppMiddleware.cs

```diff
@@ -142,8 +142,16 @@ public static class WebAppMiddlewareExtensions
         var buildRootPath = GetWebAppDistRoot(webAppProjectName, "dist");
         var templateFilePath = Path.Combine(buildRootPath, "index.html");
 
-        if (!File.Exists(templateFilePath) &&
-            Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") != "development")
+        if (Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") == "development")
+        {
+            while (!File.Exists(templateFilePath))
+            {
+                Debug.WriteLine($"Waiting for {webAppProjectName} build to to be ready...");
+                Thread.Sleep(1000);
+            }
```

Comment:
Instead of using `Thread.Sleep(1000)`, it's best practice to use `Thread.Sleep(TimeSpan.FromSeconds(1))` to avoid confusion between milliseconds and seconds.

I also strongly suggest setting a limit on the wait time. An indefinite wait might not be ideal. I have not experienced this problem, but I guess, a wait time of 10 seconds, or perhaps even just 2 seconds, should be sufficient.

```suggestion
            var tryUntil = DateTime.UtcNow.AddSeconds(10);
            while (!File.Exists(templateFilePath))
            {
                if (DateTime.UtcNow > tryUntil) break;
                Debug.WriteLine($"Waiting for {webAppProjectName} build to be ready...");
                Thread.Sleep(TimeSpan.FromSeconds(1));
            }
```